tests/test_updater_ng.py

```diff
@@ -228,6 +233,46 @@ def test_refresh_with_only_local_root(self):
     # Get targetinfo for 'file3.txt' listed in the delegated role1
     targetinfo3 = self.repository_updater.get_one_valid_targetinfo('file3.txt')
 
+
+  def test_both_target_urls_not_set(self):
+      # target_base_url = None and Updater._target_base_url = None
+      self.repository_updater = ngclient.Updater(self.client_directory,
+                                              self.metadata_url)
+      with self.assertRaises(ValueError):
+           self.repository_updater.download_target([], self.destination_directory)
+
+  def test_external_targets_url(self):
+      self.repository_updater.refresh()
+      targetinfo = self.repository_updater.get_one_valid_targetinfo("file1.txt")
+      self.repository_updater.download_target(targetinfo, self.destination_directory, self.targets_url)
+
+  def test_length_hash_mismatch(self):
+      self.repository_updater.refresh()
+      targetinfo = self.repository_updater.get_one_valid_targetinfo("file1.txt")
+      length = targetinfo.length
+      with self.assertRaises(exceptions.RepositoryError):
+        targetinfo.length = 44
+        self.repository_updater.download_target(targetinfo, self.destination_directory)
+
+      with self.assertRaises(exceptions.RepositoryError):
+        targetinfo.length = length
+        targetinfo.hashes = {"sha256": "abcd"}
+        self.repository_updater.download_target(targetinfo, self.destination_directory)
+
+  def test_updating_root(self):
+      self._modify_repository_root(lambda root: None, bump_version=True)
```

Comment:
The modification function `lambda root: None` won't do any changes on root right?
Wanted to make sure we only wanted to test here bumping the version and resigning.

Reply:
Yes, I needed a modification function doing nothing.